x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/create_security_rule_type_wrapper.ts

```diff
@@ -341,15 +338,6 @@ export const createSecurityRuleTypeWrapper: CreateSecurityRuleTypeWrapper =
             });
           }
 
-          if (!isMachineLearningParams(params) && !isEsqlParams(params)) {
```

Comment:
I think it does make sense to drop this `if` block from the shared Security wrapper, but if we do fork the `getQueryFilter` function we'll still need a broad `getFieldsForWildcard` request still in indicator match, new terms, and threshold rules. Query rules wouldn't need the broad `fieldCaps` request anymore, since they would use the new query builder function that fetches only specific fields. EQL rules don't use the fields response at all afaict so we wouldn't need it there, and ML and ESQL already don't make the field caps request. So that just leaves indicator match, new terms, and threshold, where I think we could just drop this `inputIndexFields = await getFieldsForWildcard({...` call into the beginning of those specific rule type executors.